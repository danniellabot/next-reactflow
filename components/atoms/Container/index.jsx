// create a new component from Container component in material ui
//
// Path: components/atoms/Container/index.jsx

import React from "react";
import Container from "@mui/material/Container";

const ContainerComponent = (props) => {
  const { children, className, maxWidth, disableGutters, fixed } = props;
  return (
    <Container
      className={className}
      maxWidth={maxWidth}
      disableGutters={disableGutters}
      fixed={fixed}
    >
      {children}
    </Container>
  );
};

export default ContainerComponent;
