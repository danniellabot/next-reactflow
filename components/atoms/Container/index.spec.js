// unit test for the Container component
// check max width renders correctly

import React from "react";
import { render } from "@testing-library/react";
import Container from "./index";

describe("Container", () => {
  it("should render the component", () => {
    const { container } = render(<Container />);
    expect(container).toMatchSnapshot();
  });

  it("should render the component with max width", () => {
    const { container } = render(<Container maxWidth="md" />);
    expect(container).toMatchSnapshot();
  });
});
