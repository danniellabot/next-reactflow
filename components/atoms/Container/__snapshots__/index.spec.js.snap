// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should render the component 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  />
</div>
`;

exports[`Container should render the component with max width 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthMd css-xdmu94-MuiContainer-root"
  />
</div>
`;
